### ISSUE TITLE
iOS - Headers aren't applied if body field is not specified

### ISSUE THREAD

**Issue by david-banks-roam** [MaxToyberman/react-native-ssl-pinning#13]
In the objective C code, headers are only applied if the body field exists:

```
if ([obj objectForKey:@"body"]) {
    NSDictionary * body = obj[@"body"];
    
    // this is a multipart form data request
    if([body isKindOfClass:[NSDictionary class]] && [body objectForKey:@"formData"]){
        // post multipart
        NSDictionary * formData = body[@"formData"];
        [self performMultipartRequest:manager obj:obj url:url request:request callback:callback formData:formData];
    }
    else {
          [self setHeaders:obj request:request];
        // post a string
        NSData *data = [obj[@"body"] dataUsingEncoding:NSUTF8StringEncoding];
        [request setHTTPBody:data];
        
        //TODO: if no body
    }
    
}
[self performRequest:manager request:request callback:callback ];
```

This means for GET requests, you have to explicitly set `body: ''`. Took me a few hours to figure out what was going on - our API throwing back auth errors because the Authorization header was left off the request. Headers should still be applied if body field is omitted. Or this should at least be documented.

**Comment by david-banks-roam** [comment]
I should add, specifying `body: ''` fails on Android, at least for GET requests anyway, so we need to do a platform check for iOS before setting this empty string body.

**Comment by MaxToyberman** [comment]
@david-banks-roam  The issue was solved at version 1.1.9 , can you please check it ? and reopen if you have any problem
